scripts/tools/MigrationAction.php

```diff
@@ -0,0 +1,182 @@
+                'longPrefix' => 'customParameters',
+                'required' => false,
+                'defaultValue' => [],
+                'description' => 'Custom parameters for target task ie: “option1=0\&option2=1”'
+            ],
+            'target' => [
+                'prefix' => 't',
+                'longPrefix' => 'target',
+                'required' => true,
+                'cast' => 'string',
+                'description' => sprintf(
+                    'Define unit processing task (must extend "%s")',
+                    AbstractMigrationTask::class
+                )
+            ],
+        ];
+    }
+
+    protected function provideDescription()
+    {
+    }
+
+    protected function run()
+    {
+        $startedAt = time();
+        $customParameters = [];
+
+        $chunkSize = $this->getOption('chunkSize');
+        parse_str($this->getOption('customParameters'), $customParameters);
+        $pickSize = $this->getOption('pickSize');
+        $repeat = (bool) $this->getOption('repeat');
+
+        $taskClass = $this->detectTargetClass($this->getOption('target'));
+
+        try {
+            $task = $this->spawnTask($customParameters, $chunkSize, $pickSize, $taskClass, $repeat);
+
+            $taskLogEntity = $this->getTaskLogEntity($task->getId());
+            $taskReport = $taskLogEntity->getReport();
+
+            if (0 === strcasecmp($taskLogEntity->getStatus()->getLabel(), TaskLogInterface::STATUS_FAILED)) {
+                throw new RuntimeException('task failed please refer logs');
+            }
+        } catch (Throwable $e) {
+            return common_report_Report::createFailure($e->getMessage());
+        }
+
+        $report = common_report_Report::createSuccess(
+            sprintf(
+                'Operation took %fsec and %dMb',
+                time() - $startedAt,
+                memory_get_peak_usage(true) / 1024 / 1024
+            )
+        );
+
+        if (null !== $taskReport) {
+            $report->add($taskReport);
+        }
+
+        return $report;
+    }
+
+    protected function provideUsage()
+    {
+        return [
+            'prefix' => 'h',
+            'longPrefix' => 'help',
+            'description' => 'Prints the help.',
+        ];
+    }
+
+    protected function returnJson($data, $httpStatus = 200)
+    {
+    }
+
+    private function spawnTask(
+        array $customParameters,
+        int $chunkSize,
+        int $pickSize,
+        string $taskClass,
+        bool $repeat = true
+    ): CallbackTaskInterface
+    {
+        /** @var QueueDispatcherInterface $queueDispatcher */
+        $queueDispatcher = $this->getServiceLocator()->get(QueueDispatcherInterface::SERVICE_ID);
+
+        return $queueDispatcher->createTask(
+            new $taskClass(),
+            array_merge($customParameters ,[
+                'chunkSize' => $chunkSize,
+                'pickSize' => $pickSize,
+                'repeat' => $repeat,
+            ]),
```

Comment:
```suggestion
            array_merge(
                $customParameters,
                [
                    'chunkSize' => $chunkSize,
                    'pickSize' => $pickSize,
                    'repeat' => $repeat,
                ]
            ),
```

https://www.php-fig.org/psr/psr-12/

```
Argument lists and variable lists MAY be split across multiple lines, where each subsequent line is indented once. When doing so, the first item in the list MUST be on the next line, and there MUST be only one argument or variable per line.

When the ending list (whether of arguments or variables) is split across multiple lines, the closing parenthesis and opening brace MUST be placed together on their own line with one space between them.
```

I notice we do not do it in several places and we do this kind of thing like:

```
        return new ResultFilter([
            'start' => (int) $config->getCustomParameter('start'),
            'end' => $end,
            'max' => $max
        ]);
```

The `[` should come in the next line.